RockLib.Immutable/Semimutable.cs

```diff
@@ -4,197 +4,194 @@
+      /// </remarks>
+      public Semimutable()
+          : this(default(T))
+      {
+      }
+
+      /// <summary>
+      /// Initializes a new instance of the <see cref="Semimutable{T}"/> class.
+      /// </summary>
+      /// <param name="defaultValue">The default value.</param>
+      /// <remarks>
+      /// Calls <see cref="Semimutable{T}(Func{T})"/> passing a function that returns
+      /// <paramref name="defaultValue"/> as the parameter.
+      /// </remarks>
+      public Semimutable(T? defaultValue)
+          : this(() => defaultValue)
+      {
+      }
+
+      /// <summary>
+      /// Initializes a new instance of the <see cref="Semimutable{T}"/> class.
+      /// </summary>
+      /// <param name="getDefaultValue">A function that returns the default value.</param>
+      public Semimutable(Func<T?> getDefaultValue)
+      {
+         _getDefaultValue = getDefaultValue;
+         _potentialInstance = new Lazy<T?>(getDefaultValue);
+         _lockedInstance = null;
+         HasDefaultValue = true;
+      }
+
+      /// <summary>
+      /// Gets or sets the value of the semimutable object. The setter can be called multiple times, but
+      /// only the last value "wins". Once the getter is called (or the <see cref="LockValue"/> method is
+      /// called), the value is "locked" - any value passed to the setter is ignored from this point forward.
+      /// </summary>
+      public T? Value
+      {
+         get { return GetValue(); }
+         set { SetValue(() => value); }
+      }
+
+      /// <summary>
+      /// Gets a value indicating whether this instance is locked.
+      /// </summary>
+      public bool IsLocked
+      {
+         get { return _lockedInstance != null; }
+      }
+
+      /// <summary>
+      /// Gets a value indicating whether this instance has (or will have) the default value.
+      /// </summary>
+      public bool HasDefaultValue { get; private set; }
+
+      /// <summary>
+      /// Sets the <see cref="Value"/> property to this instance's original default value.
+      /// </summary>
+      public void ResetValue()
+      {
+         SetValue(_getDefaultValue);
+      }
+
+      /// <summary>
+      /// Locks the <see cref="Value"/> property and prevents any further changes from being accepted.
+      /// </summary>
+      public void LockValue()
+      {
+         // Just call GetValue and ignore the result.
+         GetValue();
+      }
+
+      /// <summary>
+      /// Gets a method that unlocks the <see cref="Value"/> property, allowing changes to be accepted.
+      /// </summary>
+      /// <remarks>
+      /// This method should not be used "in production". Its main use is to help facilitate testing.
+      /// </remarks>
+      public MethodInfo GetUnlockValueMethod()
+      {
+         return GetType().GetMethod(nameof(UnlockValue), BindingFlags.NonPublic | BindingFlags.Instance)!;
+      }
+
+      private void UnlockValue()
+      {
+         if (_lockedInstance != null)
```

Comment:
Can this be changed to `is not null`?

Reply:
Forgot about updating null checks.

Reply:
@JasonBock is there something we can set in the .editorconfig to force this?  I did a quick search, but couldn't find anything, but might not have been searching for the right thing lol.